apps/prairielearn/src/pages/instructorGradingJob/instructorGradingJob.js

```diff
@@ -12,20 +12,19 @@ const aws = require('../../lib/aws');
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
-// NOTE: We assume that instructorGradingJob is only mounted on the course_instance
-// page route, as is currently the case. If you add a course page route, take care!
-// See instructorJobSequence, for example.
-
```

Comment:
Calling out for other reviewers: because this page is always mounted after course/instance authorization checks, I don't believe we need the complex auth checking of `instructorJobSequence`. Instead, we rely on the following:

- If a user can access a course page, they must be an instructor, and so we let them access any grading jobs associated with _just_ the course (not also a course instance)
- If a user can access a course instance instructor page, we let them view any grading jobs associated with that course instance (but not just the course, and not other course instances).

Reply:
For the presence/absence of course instance ID, I did a quick test, and it seems the course instance ID of the variant is based on the course instance in the route, even if using the instructor preview. This means that the only way the approach above would not work would be if someone explicitly changes the URL by replacing the course with course instance or vice-versa while keeping the variant ID parameter. I don't think there is any kind of link in the interface that would make this switch without also losing the variant ID or grading job ID. So I think this is a valid approach.

> If a user can access a course instance instructor page, we let them view any grading jobs associated with that course instance (but not just the course, and not other course instances).

There is a case that this should not happen, which is grading jobs created for students when the current user does not have student data viewer permission. So not "any grading jobs", but "any grading jobs without an assessment instance". This is already handled though in line 34-36, just adding it here to make sure it's clear that it has been thought through. A user without student data viewer access (but with course viewer access) can, after this change, see grading jobs without a course instance (in a route without it), or with a course instance but without an assessment instance (in a route with course instance), but not with an assessment instance. So all good from what I can see.